enhance udhcp-script and add support of ipwatchd
(should probably rewritten as user-rule or netifd module?)

diff --git a/package/network/config/netifd/files/usr/share/udhcpc/default.script b/package/network/config/netifd/files/usr/share/udhcpc/default.script
--- a/package/network/config/netifd/files/usr/share/udhcpc/default.script
+++ b/package/network/config/netifd/files/usr/share/udhcpc/default.script
@@ -48,6 +48,24 @@ case "$1" in
 	;;
 	bound)
 		setup_interface ifup
+		# restart ipwatchd if enabled
+		if [ -x /etc/init.d/ipwatchd -a /etc/init.d/ipwatchd enabled ]; then
+			/etc/init.d/ipwatchd restart
+		fi
+	;;
+	nak)
+		config_load omci
+		config_get omci_path "default" "status_file"
+		[ -f "$omci_path" ] || continue
+		config_load $omci_path
+		dev=$(echo $ifname | sed 's/\./_/g')
+		config_get dhcp_timeouts "dhcp_timeouts" "$dev"
+		if [ "$dhcp_timeouts" = "" ]; then
+			uci set $omci_path.dhcp_timeouts.$dev=1
+		else
+			dhcp_timeouts=`expr $dhcp_timeouts + 1`
+			uci set $omci_path.dhcp_timeouts.$dev=$dhcp_timeouts
+		fi
 	;;
 esac
 
